### PR TITLE
Add the type to the notification model to make it a sti model

### PR DESCRIPTION
**Pull request by danidoni**
Depends on #16222

### CODE REVIEW

File: src/api/db/migrate/20240710083919_add_type_column_to_notifications.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddTypeColumnToNotifications < ActiveRecord::Migration[7.0]
+  def change
+    add_column :notifications, :type, :string
```
Comment (krauselukas) [inline src/api/db/migrate/20240710083919_add_type_column_to_notifications.rb:3]: maybe we should add an index to the column to improve performance

Comment (danidoni) [reply src/api/db/migrate/20240710083919_add_type_column_to_notifications.rb:3]: What do you think?

File: src/api/db/schema.rb [hunk 2]
```diff
@@ -771,6 +771,7 @@
     t.boolean "rss", default: false
     t.boolean "web", default: false
     t.datetime "last_seen_at", precision: nil
+    t.string "type"
```
Comment (saraycp) [inline src/api/db/schema.rb:774]: This file is not up to date. This should include the index and a more recent date, right?

Comment (danidoni) [reply src/api/db/schema.rb:774]: Hm, you're right about the index... weird because it's present in the annotations... Let me take a look

Comment (danidoni) [reply src/api/db/schema.rb:774]: Fixed

File: src/api/db/migrate/20240710083919_add_type_column_to_notifications.rb (hunk 1 above)
Comment (krauselukas) [reply src/api/db/migrate/20240710083919_add_type_column_to_notifications.rb:3]: :+1:

**Review by krauselukas — APPROVED** [review]

**Pull request merged**